uefi-raw/src/protocol/misc.rs

```diff
@@ -22,3 +23,30 @@ pub struct TimestampProperties {
     /// example, a 24-bit counter would have an end value of `0xff_ffff`.
     pub end_value: u64,
 }
+
+/// Properties of Reset Notification.
+#[derive(Debug)]
+#[repr(C)]
+pub struct ResetNotificationProtocol {
+    pub register_reset_notify: unsafe extern "efiapi" fn(
+        this: *const Self,
+        reset_function: Option<ResetSystemFn>,
```

Comment:
I think these reset_functions don't need to be wrapped in `Option`. That would be equivalent to passing in a null pointer, and the spec for both these functions says that null is invalid for `reset_function`.

Reply:
the `Option<>` is code for `rust null` not for `c null`. It's easy to use in rust.